django/db/models/fields/__init__.py

```diff
@@ -988,41 +988,16 @@ class BooleanField(Field):
     empty_strings_allowed = False
     default_error_messages = {
         'invalid': _("'%(value)s' value must be either True or False."),
+        'invalid_nullable': _("'%(value)s' value must be either True, False, or None."),
     }
     description = _("Boolean (Either True or False)")
 
-    def __init__(self, *args, **kwargs):
-        kwargs['blank'] = True
-        super().__init__(*args, **kwargs)
-
-    def check(self, **kwargs):
-        return [
-            *super().check(**kwargs),
-            *self._check_null(**kwargs),
-        ]
-
-    def _check_null(self, **kwargs):
-        if getattr(self, 'null', False):
-            return [
-                checks.Error(
-                    'BooleanFields do not accept null values.',
-                    hint='Use a NullBooleanField instead.',
-                    obj=self,
-                    id='fields.E110',
-                )
-            ]
-        else:
-            return []
-
-    def deconstruct(self):
-        name, path, args, kwargs = super().deconstruct()
-        del kwargs['blank']
-        return name, path, args, kwargs
-
     def get_internal_type(self):
         return "BooleanField"
 
     def to_python(self, value):
```

Comment:
Should `to_python()` also have the following to mirror the behaviour from `NullBooleanField.to_python()`?
```python
        if self.null and value in ('None',):
            return None
```

Reply:
I'll look into it. That line in `NullBooleanField` is never executed by Django's test suite. It was added in 400ee0266149309ccf1d36886c4a8a0524f14efc and fixed in fc2978cc45f0fe6d596e7043dc41f4a0b56613bb.

Reply:
That is rather old...

If we don't expect to encounter `'None'`, then do we expect `'True'` and `'False'`?
It seems that these are `str()` of `None`, `True` and `False`, and would be generated by the same code?

Reply:
It's for the XML serializer which serializes `True` and `False` to strings and `None` to `<None></None>`. I'll add a test to show that the change you suggested in `to_python()` isn't needed.

Reply:
Ok.